src/store/accountStore.js

```diff
@@ -20,85 +22,105 @@ export default {
-        chain: 'tft',
-        walletName: 'savings',
-        id: '0'
-      })
 
-      // Get wallet list with names and create them all.
-      const appWallets = JSON.parse(localStorage.getItem('appWallets'))
-      if (appWallets != null && appWallets) {
-        for (const appWallet of appWallets.filter(
-          x => x.doubleName === userData.doubleName
-        )) {
-          appWallet.id = 0
-          context.dispatch('createWallet', appWallet)
-        }
-      } else if (appWallets == null) {
-        // Loop 20 wallets and check balance
-        context.commit('setImportingWallets', true)
-        for (let index = 3; index <= 20; index++) {
-          const appWallet = {
-            chain: 'tft',
-            id: '0',
-            walletName: `Wallet${index}`,
-            doubleName: userData.doubleName
-          }
-          context.dispatch('createWallet', appWallet)
-        }
-        context.dispatch('updateAccounts', () => {
-          setTimeout(() => {
-            var wallets = context.getters.wallets.filter(x => !(x && x.holder && x.holder.type && x.holder.type === 'imported'))
-            console.log(`wallets`, wallets.length)
-            for (let index = wallets.length - 1; index > 1; index--) {
-              var wallet = wallets[index]
-              if (!wallet.transaction || !wallet.transaction.length) {
-                context.getters.accounts.forEach(account => {
-                  if (account.type && account.type !== 'imported') {
-                  account.wallet_delete(index, wallet.name)
-                  }
-                })
-              } else {
-                for (
-                  let existingWalletIndex = 0;
-                  existingWalletIndex < index;
-                  existingWalletIndex++
-                ) {
-                  const newWallet = wallets[existingWalletIndex]
-                  var postMsg = {
-                    type: 'ADD_APP_WALLET',
-                    walletName: newWallet.name,
-                    doubleName: userData.doubleName
-                  }
+      context.commit('setImportingWallets', true)
 
-                  // Print.postMessage(JSON.stringify(postMsg))
-                }
-                context.commit('setImportingWallets', false)
-                return
-              }
-            }
+      window.account = tfAccount
 
+      //  daily and savings are always generated
+      await tfAccount.wallet_new(`daily`, tfAccount.wallet_count, 1)
+      await tfAccount.wallet_new(`savings`, tfAccount.wallet_count, 1)
+      await tfAccount.update_account()
+      
+      await context.dispatch('loadAppWallets', tfAccount)
+      console.log('finished loading appwallets')
+      context.commit('setAccounts', [tfAccount])
+      
+      await context.dispatch('loadImportedWallets')
 
-            var pstMsg = {
-              type: 'ADD_APP_WALLET'
-            }
+      context.commit('setImportingWallets', false)
+      router.push({ name: 'home' })
```

Comment:
Just my thoughts, login souldn't be responsible to redirect, this could be moved up in the call stack